File: cpp_sim/188623/Mlecz.cpp
#include "Mlecz.h"

Mlecz::Mlecz(Swiat* swiatOrganizmu, int posX, int posY, int age, bool czyCzeka)
: Roslina(swiatOrganizmu, 0, posX, posY, age, czyCzeka){
    
};

void Mlecz::rysowanie(){
    printf("M");
};

string Mlecz::getNazwa(){
    return ("Mlecz");
};

void Mlecz::akcja(){
    // Zwiększenie wieku organizmu
    wiek++;
    // Wykonuje trzy próby rozmnożenia się
    Roslina::akcja();
    Roslina::akcja();
    Roslina::akcja();
};

Organizm *Mlecz::klonowanie(int posX, int posY){
    return new Mlecz(swiatOrganizmu, posX, posY, 1, true);
};

Mlecz::~Mlecz(){
    
};

File: java_sim/simulation_188623/src/Mlecz.java
import javax.swing.ImageIcon;

public class Mlecz extends Roslina {

    Mlecz(Swiat swiatOrganizmu, int x_pos, int y_pos, int wiek, boolean czyCzeka) {
        super(swiatOrganizmu, 0, x_pos, y_pos, wiek, czyCzeka);
        this.ikonka = new ImageIcon("zdjecia/mlecz.png");      
        this.hexIkonka = new ImageIcon("zdjecia/mleczHex.png");  
    }
    
    String getNazwa() {
        return ("Mlecz");
    }

    @Override
    char getZnaczek() {
        return '6'; 
    }

    void akcja() {
        wiek++;
        //trzy proby rozmnozenia sie
        super.akcja();
        super.akcja();
        super.akcja();
    }

    @Override
    Organizm klonowanie(int posX, int posY) {
        return new Mlecz(swiatOrganizmu, posX, posY, 1, true);
    }
}
